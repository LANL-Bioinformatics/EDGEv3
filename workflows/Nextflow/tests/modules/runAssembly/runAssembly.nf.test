nextflow_workflow {

    name "Test Workflow ASSEMBLY"
    script "modules/runAssembly/runAssembly.nf"
    workflow "ASSEMBLY"

    test("IDBA") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testAssembly"
                }
                assembly{
                    assembler = "IDBA_UD"
                }
            }
            workflow {
                """
                input[0] = params.assembly.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = channel.of("151")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.outContigs.find { file(it).name == "testAssembly_contigs.fa" }).exists()},
                {assert file(workflow.out.annotationContigs.find { file(it).name == "testAssembly_contigs_700up.fa" }).exists()}
            ) 
        }

    }

    test("SPAdes") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testAssembly"
                }
                assembly{
                    assembler = "spades"
                }
            }
            workflow {
                """
                input[0] = params.assembly.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = channel.of("151")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.outContigs.find { file(it).name == "testAssembly_contigs.fa" }).exists()},
                {assert file(workflow.out.annotationContigs.find { file(it).name == "testAssembly_contigs_700up.fa" }).exists()}
            ) 
        }

    }


    test("LRASM") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testAssembly"
                }
                assembly{
                    assembler = "lrasm"
                    lrasm {
                        ec = true
                    }
                }
            }
            workflow {
                """
                input[0] = params.assembly.plus(params.shared)
                input[1] = [file("${projectDir}/nf_assets/NO_FILE")]
                input[2] = file("${projectDir}/test_data/reads.fastq.gz")
                input[3] = channel.of("151")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.outContigs.find { file(it).name == "testAssembly_contigs.fa" }).exists()},
                {assert file(workflow.out.annotationContigs.find { file(it).name == "testAssembly_contigs_700up.fa" }).exists()}
            ) 
        }

    }


    test("Unicycler") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testAssembly"
                }
                assembly{
                    assembler = "Unicycler"
                }
            }
            workflow {
                """
                input[0] = params.assembly.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = channel.of("151")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.outContigs.find { file(it).name == "testAssembly_contigs.fa" }).exists()},
                {assert file(workflow.out.annotationContigs.find { file(it).name == "testAssembly_contigs_700up.fa" }).exists()}
            ) 
        }

    }


    test("MEGAHIT") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testAssembly"
                }
                assembly{
                    assembler = "megahit"
                }
            }
            workflow {
                """
                input[0] = params.assembly.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = channel.of("151")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success},
                {assert file(workflow.out.outContigs.find { file(it).name == "testAssembly_contigs.fa" }).exists()},
                {assert file(workflow.out.annotationContigs.find { file(it).name == "testAssembly_contigs_700up.fa" }).exists()}
            )
        }

    }

}
