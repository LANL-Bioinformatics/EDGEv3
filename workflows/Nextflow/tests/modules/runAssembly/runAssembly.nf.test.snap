{
    "SPAdes": {
        "content": [
            {
                "0": [
                    "testAssembly_contigs.fa:md5,b205c190c365337380dee8a3bcc535e4"
                ],
                "1": [
                    "testAssembly_contigs_700up.fa:md5,149b7cd8d5110b39c33c284429c5f086"
                ],
                "annotationContigs": [
                    "testAssembly_contigs_700up.fa:md5,149b7cd8d5110b39c33c284429c5f086"
                ],
                "outContigs": [
                    "testAssembly_contigs.fa:md5,b205c190c365337380dee8a3bcc535e4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-25T13:45:36.26872"
    },
    "IDBA": {
        "content": [
            {
                "0": [
                    "testAssembly_contigs.fa:md5,03db7c3078458dd0ca091fa0662b140e"
                ],
                "1": [
                    "testAssembly_contigs_700up.fa:md5,aa162c4f9a3684c7fcfd55dd044de22c"
                ],
                "annotationContigs": [
                    "testAssembly_contigs_700up.fa:md5,aa162c4f9a3684c7fcfd55dd044de22c"
                ],
                "outContigs": [
                    "testAssembly_contigs.fa:md5,03db7c3078458dd0ca091fa0662b140e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-25T13:33:59.338422"
    }
}